lib/pharos/configuration/host.rb

```diff
@@ -93,16 +93,44 @@ def api_address
         api_endpoint || address
       end
 
+      # @return [String]
       def peer_address
         private_address || private_interface_address || address
       end
 
+      # @param host [Pharos::Configuration::Host]
+      # @return [String]
+      def peer_address_for(host)
+        if region == host.region
+          peer_address
+        else
+          address
+        end
+      end
+
+      # @return [String]
+      def region
+        labels['failure-domain.beta.kubernetes.io/region'] || 'unknown'
+      end
+
+      # @return [Hash]
       def labels
-        return @attributes[:labels] unless worker?
+        labels = @attributes[:labels] || {}
+
+        labels['node-address.kontena.io/external-ip'] = address
+        labels['node-role.kubernetes.io/worker'] = '' if worker?
+
+        labels
+      end
 
-        @attributes[:labels] || { 'node-role.kubernetes.io/worker': "" }
+      # @return [Hash]
+      def checks
+        @checks ||= {}
```

Comment:
I think you'll need to change `GatherFacts` `@host.checks = host_checks` to  `@host.checks.replace(host_checks)`

Reply:
I don't think so... because of `attr_accessor :checks` .

Reply:
Right. I removed the accessor since I added the reader manually.